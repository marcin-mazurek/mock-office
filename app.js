const express = require('express');
require('./dist/double-rest-api');

const serve = express.static('./dist/double-gui');
const app = express();
const server = app.use(serve);

server.listen(3070, () => {
  console.log('GUI address: http://127.0.0.1:3070');
});
