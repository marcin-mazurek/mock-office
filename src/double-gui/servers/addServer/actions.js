export const INIT = 'addServer/INIT';

export default (name, port, serverType, isSecure, keyPath, certPath) => ({
  type: INIT,
  name,
  port,
  serverType,
  isSecure,
  keyPath,
  certPath
});
