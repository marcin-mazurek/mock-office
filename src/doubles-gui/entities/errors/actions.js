export const ADD = 'errors/ADD';
export const REMOVE = 'errors/REMOVE';

export const add = reason => ({
  type: ADD,
  reason
});

export const remove = id => ({
  type: REMOVE,
  id
});
