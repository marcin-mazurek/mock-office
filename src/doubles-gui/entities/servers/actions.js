export const ADD = 'servers/ADD';
export const SELECT = 'servers/SELECT';
export const START = 'servers/START';
export const STOP = 'servers/STOP';
export const REMOVE = 'servers/REMOVE';
export const RENAME = 'servers/RENAME';

export const add = (name, port, serverType, id, secure) => ({
  type: ADD,
  name,
  port,
  id,
  serverType,
  secure
});

export const select = id => ({
  type: SELECT,
  id
});

export const start = id => ({
  type: START,
  id
});

export const stop = id => ({
  type: STOP,
  id
});

export const remove = id => ({
  type: REMOVE,
  id
});

export const rename = (id, name) => ({
  type: RENAME,
  id,
  name
});
