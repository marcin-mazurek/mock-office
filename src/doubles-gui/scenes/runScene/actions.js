export const RUN = 'scenes/RUN';
export const STOP = 'scenes/STOP';

export const run = (serverId, taskId) => ({
  type: RUN,
  serverId,
  taskId
});

export const stop = (serverId, taskId) => ({
  type: STOP,
  serverId,
  taskId
});
