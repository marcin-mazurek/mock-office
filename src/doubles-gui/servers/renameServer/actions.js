export const INIT = 'renameServer/INIT';

export const init = (id, name) => ({
  type: INIT,
  id,
  name
});
