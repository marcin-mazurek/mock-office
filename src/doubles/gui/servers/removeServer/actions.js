export const INIT = 'removeServer/INIT';

export const init = id => ({
  type: INIT,
  id
});
