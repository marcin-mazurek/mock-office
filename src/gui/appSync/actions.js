export const RESTORE_STATE = 'appSync/RESTORE_STATE';
export const restoreStateAction = servers => ({
  type: RESTORE_STATE,
  servers
});
