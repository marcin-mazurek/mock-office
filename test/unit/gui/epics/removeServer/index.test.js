import { fromJS } from 'immutable';
import configureMockStore from 'redux-mock-store';
import { createEpicMiddleware } from 'redux-observable';
import { Scenario, Server } from '../../../../../src/lib/gui/app/entities';
import { removeButtonClickedAction } from '../../../../../src/lib/gui/components/InspectServer/actions';

describe('removeServerEpic', () => {
  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
  });

  test.only('success path actions snapshot', (done) => {
    jest.mock('../../../../../src/lib/gui/resources/api', () => ({
      removeServer() {
        return Promise.resolve({
          data: {
            id: 'server-id'
          }
        });
      }
    }));
    const removeServerEpic = require('../../../../../src/lib/gui/epics/removeServer').default;
    const epicMiddleware = createEpicMiddleware(removeServerEpic);
    const mockStore = configureMockStore([epicMiddleware]);
    const store = mockStore(fromJS({
      sidebar: {
        currentDisplayedServerId: ''
      }
    }));
    store.dispatch(removeButtonClickedAction('server-id'));
    setTimeout(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    }, 100);
  });

  test('fail path actions snapshot', (done) => {
    jest.mock('../../../../../src/lib/gui/resources/api', () => ({
      removeServer() {
        return Promise.resolve({
          error: 'Error message'
        });
      }
    }));
    const removeServerEpic = require('../../../../../src/lib/gui/epics/removeServer').default;
    const epicMiddleware = createEpicMiddleware(removeServerEpic);
    const mockStore = configureMockStore([epicMiddleware]);
    const store = mockStore(fromJS({
      servers: {
        entities: {
          'server-id': new Server({ id: 'server-id', scenario: 'scenario-id' })
        },
        ids: ['server-id']
      },
      scenarios: {
        entities: {
          'scenario-id': new Scenario({ id: 'scenario-id' })
        },
        ids: ['scenario-id']
      },
      mocks: {
        entities: {},
        ids: []
      }
    }));
    store.dispatch(removeButtonClickedAction('server-id'));
    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });
});
