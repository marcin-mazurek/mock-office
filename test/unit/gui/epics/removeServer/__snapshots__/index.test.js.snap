// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeServerEpic fail path actions snapshot 1`] = `
Array [
  Object {
    "id": "server-id",
    "type": "InspectServer/REMOVE_BUTTON_CLICKED",
  },
  Object {
    "reason": "Error message",
    "type": "removeServer/FAILED",
  },
]
`;

exports[`removeServerEpic success path actions snapshot 1`] = `
Array [
  Object {
    "id": "server-id",
    "type": "InspectServer/REMOVE_BUTTON_CLICKED",
  },
  Object {
    "id": "server-id",
    "type": "removeServer/SUCCEEDED",
  },
]
`;
